src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -271,7 +271,8 @@ public static Assembly LoadFile(string path)
         }
 
         [RequiresUnreferencedCode("Types and members the loaded assembly depends on might be removed")]
-        [RequiresAssemblyFiles(Message = "Calling 'System.Reflection.Assembly.LoadFromResolveHandler' will return a null assembly for assemblies embedded in a single-file app")]
+        [UnconditionalSuppressMessage("SingleFile", "IL3000:Avoid accessing Assembly file path when publishing as a single file",
+            Justification = "requestingAssembly contains a path to the assembly location outside of the single file application")]
```

Comment:
This is the case where we should never get here for an assembly from single-file bundle, right? I would reword it slightly to make that more obvious.

Reply:
Yeah, since it gets called by LoadFrom(string) we load from a string with a location outside the bundle, will change the wording